optimade/validator/validator.py

```diff
@@ -1285,6 +1290,12 @@ def _test_page_limit(
                     "Endpoint suggested more data was available but provided no valid links->next link."
                 )
 
+            if next_link in previous_links:
+                raise ResponseError(
+                    f"The next link {next_link} has been provided already for a previous page."
+                )
+            previous_links.add(next_link)
+
```

Comment:
Previously, it was valid to return the same `next` link for every page and get stuck in a cycle